Dispatcher metadata

Jurisak opened the pull request:
I've added simple ci_message data extractor which also includes some initial tests and .travis.yaml

Commit:
ci_message data extractor first implementation

This plugin returns json file with important data from ci_message.
This commit contains simple test data.

Commit:
ci_message data extractor first implementation

This plugin returns json file with important data from ci_message.
This commit contains simple test data.

Added travis.yml for Travis CI testing

Commit:
Added travis.yml for Travis CI testing

Commit:
Added travis.yml for Travis CI testing

Commit:
Added travis.yml for Travis CI testing

Commit:
Added travis.yml for Travis CI testing

Commit:
Merge branch 'dispatcher_metadata' of github.com:Jurisak/metamorph into dispatcher_metadata

gsr-shanks commented:
Why Travis CI ?

gsr-shanks commented on lib/logging_conf.py:
```diff
@@ -27,3 +27,8 @@ def setup_logging(default_path='logging.json', default_level=logging.INFO, env_k
         json.dump(dict(messages=input_data), metamorph, indent=2)
+
+
+def read_json_file(input_file):
```
This should be in a different file. Don't think it should be in 'logging_conf.py'.

Jurisak replied:
ok

Jurisak commented:
Do you have any other better CI tool? I've worked with Travis CI so I chose it again.

gsr-shanks commented:
Github itself has a CI.

Jurisak commented:
I never heard about that. Do you have any installation guides or documents?

gsr-shanks commented:
Never mind, I was thinking about Gitlab when I wrote that comment. Travis CI is fine for the moment.

gsr-shanks commented:
@Jurisak Just checking if you need anything here. Waiting for your patch for my review comment.

Jurisak commented:
@gsr-shanks Expect the patch today.

Commit:
logging_conf renamed to support_functions

logging_conf contain multiple functions which are not related to
logging. Therefore this file was renamed.

gsr-shanks commented:
Please add docstrings.

Commit:
Documentation for  message data extractor

gsr-shanks commented:
Thanks for the documentation. Could you please add docstrings as well in plugins/message_data_extractor.py

Commit:
Added docstrings

Jurisak commented:
Should I create an ansible plugin from plugins/message_data_extractor.py as well?

gsr-shanks commented:
Doesn't seem to work with https://url.corp.redhat.com/469025 message.
Do we need is_closed_build(), seems redundant ?

Yes, please create an ansible plugin as well.

Jurisak commented:
@gsr-shanks  so https://url.corp.redhat.com/469025 is correct ci-message?

gsr-shanks commented:
Yes and https://url.corp.redhat.com/7f61c2a is the job.

Jurisak commented:
Do you have any different ci-messages? I've started with ci-message which contains header and message but yours is not.

gsr-shanks commented:
I don't have any other. I think we should check with Daniel or someone from RCM for the types of CI message.

Jurisak commented:
OK I'll do that...

Commit:
Added ansible plugin for message_data_extractor

- Still missing support for other CI-messages

gsr-shanks commented:
@Jurisak This PR is waiting on your discussion with RCM right ?

Jurisak commented:
Yes. I'm investigating CI-messages. I'll let you know.

gsr-shanks commented:
@Jurisak This PR has conflicts. Please have them resolved.

Jurisak commented:
@gsr-shanks Sure I'll solve them as soon as possible.

Commit:
Merge branch 'master' of github.com:Jurisak/metamorph into dispatcher_metadata

Jurisak commented:
@robnester-rh Could you do a code review here as well? Thanks.

robnester-rh commented on metamorph/library/message_data_extractor.py:
```diff
@@ -0,0 +1,148 @@
+    def read_input_file(self):
+        try:
+            self.ci_message = read_json_file(self.ci_message_file)
+        except Exception as detail:
```
Generally, I try to avoid such a wide Exception catch as the only one. 
It might be better to catch exactly what you're looking for. In this case, probably FileNotFoundError, and then probably something like json.decoder.JSONDecodeError.
```

robnester-rh commented on metamorph/lib/support_functions.py:
```diff
@@ -27,3 +27,8 @@ def setup_logging(default_path='logging.json', default_level=logging.INFO, env_k
 def storing_pretty_json(input_data, output="metamorph.json"):
```
just a suggestion: rename to write_json_file() to follow with read_json_file() below

robnester-rh commented on metamorph/library/message_data_extractor.py:
```diff
@@ -0,0 +1,148 @@
+from metamorph.lib.support_functions import setup_logging, read_json_file, storing_pretty_json
+import logging
+import json
+from ansible.module_utils.basic import AnsibleModule
```
suggestion: separate std library, 3rd party, and project local imports with a blank line.

Jurisak replied:
OK

Commit:
Code refactorization after code review

gsr-shanks commented:
Please resolve conflicting files.

gsr-shanks commented:
For this PR, we have a ci-message which contains header and message and other one doesn't. We should accommodate both use cases and merge this. 

Any further ci_message can be supported upon request.

Commit:
Merge branch 'master' of github.com:Jurisak/metamorph into dispatcher_metadata

Commit:
import fix in resultsdb files. from logging_conf -> support_functions

Jurisak commented:
@gsr-shanks Could you please send me the second ci-message type?
Thanks

Commit:
Added tests for dispatcher plugin.

+ ansible plugin tests

Commit:
Inserted requirements.txt file support
